### ISSUE TITLE
Belegnummer (Verknüpfung Kassen-DEP und RKSV-DEP)

### ISSUE THREAD

**Issue by ggoelles** [a-sit-plus/at-registrierkassen-mustercode#416]
Wie genau erfolgt die Verknüpfung des Kassen-DEP und des RKSV-DEP?
Wir verwenden derzeit eine fortlaufende Nummer für alle Belegarten (Barumsatz, Barentnahme und -einlage).
Im maschinenlesbaren Code hat eine eindeutige, fortlaufende Belegnummer aufzuscheinen.
Nachdem ich hier die Issues bezüglich der Belegnummer gelesen habe signieren manche alle Belege, manche führen einen eigenen Nummernkreis.
Wenn ich jetzt eine externe Belegnumme und interne fortlaufende Barbelegnummer habe:
Wie geht der Prüfer im Rahmen einer Prüfung vor bzw. verknüpft das Kassen-DEP mit den Umsätzen aus dem RKSV-DEP?

**Comment by ErichFreitag** [comment]
z.B. indem sie im Kassen-DEP die Nummer des RKSV-Barumsatzes mitschreiben. Alternativ z.B. über Datum und Uhrzeit, das kann aber schon wieder geringfügig abweichen.

**Comment by ggoelles** [comment]
Ok danke. Also intern einfach mitprotokollieren.

**Comment by ErichFreitag** [comment]
Ja, dann sind die Zusammenhänge eindeutig und klar und leicht zuordenbar.

**Comment by Flanelli** [comment]
@ErichFreitag 
Wenn Datum+Uhrzeit eines Beleges im RKSV-DEP und Kassen-DEP nicht 100% exakt übereinstimmen
wurde meines Erachtens in der Programmierung ein gravierender Fehler gemacht und zwar jener
dass die Uhrzeit bei jedem Schreibvorgang ( oder auch Druckvorgang ) neu von der aktuellen Systemzeit
eingelesen wird anstatt die Uhrzeot des "allerersten" Vorganges der Belegerstellung in alle anderen Vorgänge dieses Beleges sozusagen "mitzuschleppen".
Annahme:
1: Vorgang = Estellen der Signatur: 12:20:45
danach z.B.:
Speichern im RKSV-DEP 0,5 Sekunden später: Es wird zwangsläufig die Zeit übereinstimmen
da der Wert ja in den Signaturvorgang übernommen wurde.
Extrahieren des Q-Codes 0,5 Sekunden später: Es wird zwangsläufig die Zeit übereinstimmen
da der Wert ja in der Kette vorhanden ist.

Druck 1 Sekunde später: Am Beleg mu0 die Zeit des ersten Vorganges stehen
andernfalls würde die Zeit am Beleg mit der Zeit im QR-Code nicht übereintsimmen.

Druckwiederholung 20 Sekunden später : Am Beleg mu0 die Zeit des ersten Vorganges stehen
andernfalls würde die Zeit am Beleg mit der Zeit im QR-Code nicht übereintsimmen.

Spichern im kassen-DEP 21 Sekunden später: Es mu0 die Zeit des ersten Vorganges stehen 
andernfalls würde die Zeit mit jener des Belegs nicht übereinstimmen.

Klingt möglicherweise etwas spitzfindig aber ein Beleg kann eben nur eine einzige Zeit in allen
Facetten aufweisen es sei denn man druckt separat ausgewiesen die echte Druckzeit an.
z.B.
Erstellt am 12.03.2017 um 12:20:45
Gedruckt am 12.03.2017 um 12:21:05

**Comment by ErichFreitag** [comment]
(Datum und) die Uhrzeit am Beleg und im maschinenlesbaren Code und damit auch im RKSV-DEP muss die selbe sein, da bin ich mit ihnen.

Mit welchem Zeitstempel allerdings der Geschäftsfall im Kassenjournal abgelegt ist wird sehr implementierungabhängig sein und muss meiner Meinung nach auch nicht 100% mit der Belegausstellung ident sein. Wichtig ist, dass man die Vorgänge einfach zusammenfinden kann, beispielsweise indem man den maschinenlesbaren Code mit in das Journal schreibt, zumindest aber dessen Belegnummer und Datum/Uhrzeit.

**Comment by Flanelli** [comment]
Selbstverständlich haben Sie recht Hr. Freitag
Da habe ich mich etwas unglücklich ausgedrückt mit dem Zeitstempel des Kassenjournales
Natürloich meinte ich nicht den physisch identen Zeitstempel sondern den inhaltlichen so wie sie sides
auch angeführt haben.

Beispiel: DEP - BELEGDETAIL
Physisch gespeichert am 2017-03-12 um 09:36:**30**
Erstellungszeitpunkt Beleg  am 2017-03-12 um 09:36:**18**
{ "Belege-Detail": [
 {   "Belegnummer": "100235",
   "Kassen-ID": "KASSA1",
   "Status SE": "OK",
   "QR-CODE": "_R1-AT2_KASSA1_100235_2017-03-04T09:36:**18**_563,80_0,00_0,00_0,00_0,00_usw.. 
   "Bezug-POS": "113751",
   "Beleg-Datum-Uhrzeit": "2017-03-12T09:36:**18**",
   "Beleg-Summe": 563,80,
   "Beleg-Inhalte": [
     {
       "Artikel": "01010001 MANTEL MARIN 42",
       "Menge": 1.00,
       "Preis": 326.50
     },
     {
       "Artikel": "01050007 MANTEL KURZ 40",
       "Menge": 1.00,
       "Preis": 237.30
     }
   ]
}

**Comment by ErichFreitag** [comment]
Immer habe ich auch nicht recht, aber da sind wir uns einig.
